plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/handlers/AnalysisKickOff.java

```diff
@@ -89,9 +91,12 @@ public boolean setUp(final IJavaElement iJavaElement) {
 			Activator.getDefault().logInfo("JavaCore could not create IJavaProject for project " + ip.getName() + ".");
 			return false;
 		}
-		this.curProj = javaProject;
-		return true;
-	}
+		
+		KotlinUtils.compileKotlinFiles(ip);

```

Comment:
Do not call the kotlin plugin from inside this one like this. This call causes the kotlin plugin to be required under all circumstances.

Reply:
Could you please suggest me some way to achieve this? I've implemented rest of your review changes except this one.

Reply:
I think reflection or some kind of dependency injection would be the way to go. Reflection is not that clean.  We should look up some dependency injection frameworks that are clean. This sounds tricky to accomplish but a needed requirement as we do not want to have dependency from kotlin to base.

Reply:
One option is to use the factory pattern. Have a base factory runner that calls the kotlin runner if configuration is kotlin. https://www.tutorialspoint.com/design_pattern/factory_pattern.htm